rbx_binary/src/deserializer.rs

```diff
@@ -947,6 +974,39 @@ impl<R: Read> BinaryDeserializer<R> {
                     });
                 }
             },
+            Type::SharedString => match canonical_type {
+                VariantType::SharedString => {
+                    let mut values = vec![0; type_info.referents.len()];
+                    chunk.read_interleaved_u32_array(&mut values)?;
+
+                    for (value, referent) in values.into_iter().zip(&type_info.referents) {
+                        let shared_string =
+                            self.shared_strings.get(value as usize).ok_or_else(|| {
+                                InnerError::InvalidPropData {
+                                    type_name: type_info.type_name.clone(),
+                                    prop_name: prop_name.clone(),
+                                    valid_value: "a valid SharedString",
+                                    actual_value: format!("{:?}", value),
+                                }
+                            })?;
+
+                        let instance = self.instances_by_ref.get_mut(referent).unwrap();
+
+                        instance.builder.add_property(
+                            &canonical_name,
+                            Variant::SharedString(shared_string.clone()),
```

Comment:
```suggestion
                            shared_string.clone(),
```

Should be able to avoid naming `Variant` here explicitly.